src/components/editor/ResourceTemplateForm.jsx

```diff
@@ -85,7 +83,13 @@ export class ResourceTemplateForm extends Component {
 
   resourceTemplatePromises = async (templateRefs) => {
     return Promise.all(templateRefs.map(rtId =>
-      getResourceTemplate(rtId)
+      getResourceTemplate(rtId).catch(err => {
+        const errId = decodeURIComponent(/[^/]*$/.exec(err.url)[0])
+        this.setState({tempErrorUrl: errId})
```

Comment:
I have seen this pattern used elsewhere in the editor and I'm a bit puzzled. As far as I can tell, the only code that looks for and uses `tempErrorUrl` in state is two lines beneath. Which makes me wonder: what do we get out of injecting this into application state? In other words, how is this code different from:

```javascript
const joinedErrorUrls = this.state.templateErrors.slice(0)
joinedErrorUrls.push(errId)
this.setState({templateErrors: joinedErrorUrls})
```

:question:

Reply:
That works fine in this case and I fixed it up. I'd have to take a second look at the other instances where I used this pattern. I think here I was blindly following the same pattern I used elsewhere and didn't need to, but I think maybe elsewhere it was used for a reason (maybe because of updating state or awaiting promises). I remember struggling with getting the array in question populated correctly and that pattern is what ended up working for me. In any case, It's definitely work taking a second look at those, and you may already be refactoring the components where those patterns were anyway.